src/main/java/forestry/core/genetics/Chromosome.java

```diff
@@ -22,140 +22,161 @@
-		if (primary.isDominant()) {
-			return primary;
-		}
-		if (secondary.isDominant()) {
-			return secondary;
-		}
-		// Leaves only the case of both being recessive
-		return primary;
+		return active;
 	}
 
 	@Override
 	public IAllele getInactiveAllele() {
-		if (!secondary.isDominant()) {
-			return secondary;
-		}
-		if (!primary.isDominant()) {
-			return primary;
-		}
-		// Leaves only the case of both being dominant
-		return secondary;
+		return inactive;
 	}
 
-	/* HELPER FUNCTIONS */
-	public static IChromosome inheritChromosome(Random rand, IChromosome parent1, IChromosome parent2) {
-
-		IAllele choice1;
+	@Override
+	public IChromosome inheritChromosome(Random rand, IChromosome other) {
+		IAllele firstChoice;
 		if (rand.nextBoolean()) {
-			choice1 = parent1.getPrimaryAllele();
+			firstChoice = getActiveAllele();
 		} else {
-			choice1 = parent1.getSecondaryAllele();
+			firstChoice = getInactiveAllele();
 		}
 
-		IAllele choice2;
+		IAllele secondChoice;
 		if (rand.nextBoolean()) {
-			choice2 = parent2.getPrimaryAllele();
+			secondChoice = other.getActiveAllele();
 		} else {
-			choice2 = parent2.getSecondaryAllele();
+			secondChoice = other.getInactiveAllele();
 		}
 
 		if (rand.nextBoolean()) {
-			return new Chromosome(choice1, choice2);
+			return Chromosome.create(firstChoice, secondChoice);
 		} else {
-			return new Chromosome(choice2, choice1);
+			return Chromosome.create(secondChoice, firstChoice);
 		}
 	}
 
 	@Override
 	public String toString() {
-		return "{" + primary + ", " + secondary + "}";
+		return "{" + active + ", " + inactive + "}";
+	}
+
+	/* HELPER FUNCTIONS */
+	@Deprecated
+	public static IChromosome inheritChromosome(Random rand, IChromosome firstParent, IChromosome secondParent) {
+		return firstParent.inheritChromosome(rand, secondParent);
+	}
+
+	@Nullable
+	static IAllele getActiveAllele(NBTTagCompound chromosomeNBT){
+		String alleleUid = chromosomeNBT.getString(Chromosome.ACTIVE_ALLELE_TAG);
+		return AlleleManager.alleleRegistry.getAllele(alleleUid);
+	}
+
+	@Nullable
+	static IAllele getInactiveAllele(NBTTagCompound chromosomeNBT){
+		String alleleUid = chromosomeNBT.getString(Chromosome.ACTIVE_ALLELE_TAG);
+		return AlleleManager.alleleRegistry.getAllele(alleleUid);
+	}
+
+	/**
+	 * Checks if the allele is not null and valid for that type of chromosome.
+	 */
+	private static IAllele validateAllele(@Nullable String speciesUid, IChromosomeType chromosomeType, @Nullable IAllele allele) {
+		if (!chromosomeType.getAlleleClass().isInstance(allele)) {
+			ISpeciesRoot speciesRoot = chromosomeType.getSpeciesRoot();
+
+			IAllele[] template = null;
+
+			if (speciesUid != null) {
+				template = speciesRoot.getTemplate(speciesUid);
+			}
+
+			if (template == null) {
+				template = speciesRoot.getDefaultTemplate();
+			}
+
+			return template[chromosomeType.ordinal()];
+		}
+		return allele;
+	}
+
+	private static IAllele getActiveAllele(IAllele firstAllele, IAllele secondAllele) {
```

Comment:
This strategy is good but it may not be random enough.
We should mix them randomly if both are dominant or recessive. If the `firstAllele` always comes from the queen and the `secondAllele` always comes from the mate, it could allow the player to control the outcome.

Reply:
Do we not mix them randomly with `inheritChromosome()`?

Reply:
You're right, ok.